flytekit/core/artifact.py

```diff
@@ -273,6 +273,24 @@ def __init__(
         self.reference_artifact: Optional[Artifact] = None
         self.granularity = granularity
 
+    def __rich_repr__(self):
```

Comment:
I see that this is used by `_repr_html_`, but is there a use case where `__rich_repr__` is used directly by `rich`?

Reply:
Yeah, in fact this is a protocol that rich [proposes](https://rich.readthedocs.io/en/stable/pretty.html#rich-repr-protocol) and expects that will become more available in the future. From the rich docs:
```
Note that you can add __rich_repr__ methods to third-party libraries without including Rich as a dependency. If Rich is not installed, then nothing will break. Hopefully more third-party libraries will adopt Rich repr methods in the future.
```